dialog.go

```diff
@@ -229,13 +263,14 @@ func CreateSelectDialog(title string, items []string, selectedItem int, typ Sele
 	btn2 := CreateButton(frm1, AutoSize, AutoSize, "Cancel", Fixed)
 	btn2.OnClick(func(ev Event) {
 		dlg.result = DialogButton2
+		dlg.edtResult = ""
 		dlg.value = -1
 		WindowManager().DestroyWindow(dlg.View)
 		if dlg.onClose != nil {
 			dlg.onClose()
 		}
 	})
-	ActivateControl(dlg.View, btn2)
+	ActivateControl(dlg.View, dlg.edit)
```

Comment:
One more fix to go. have you tested it with list/radio dialog type? It seems that `dlg.edit` is null here if `typ != SelectDialogEdit`. As I think, it should activate btn2 or edit depending on dialog type.